docs/user/openstack/vgpu.md

```diff
@@ -0,0 +1,456 @@
+```
+
+
+## Create vGPU enabled Worker Nodes
+
+Worker nodes can be created by using machine API. To do that,
+[create new machineSet in OpenShift](https://docs.openshift.com/container-platform/4.11/machine_management/creating_machinesets/creating-machineset-osp.html).
+
+```console
+$ oc get machineset -n openshift-machine-api <machineset_name> -o yaml > vgpu_machineset.yaml
+```
+
+Edit yaml file, be sure to have different name, have replicas set to the amount
+of your cGPU capacity at maximum and set the right flavor, which would hint
+OpenStack about right resources to include into virtual machine:
+
+```yaml
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  annotations:
+    machine.openshift.io/memoryMb: "8192"
+    machine.openshift.io/vCPU: "4"
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+    machine.openshift.io/cluster-api-machine-role: <node_role>
+    machine.openshift.io/cluster-api-machine-type: <node_role>
+  name: <infrastructure_ID>-<node_role>-gpu-0
+  namespace: openshift-machine-api
+spec:
+  replicas: <amount_of_nodes_with_gpu>
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>-gpu-0
+  template:
+    metadata:
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_ID>
+        machine.openshift.io/cluster-api-machine-role: <node_role>
+        machine.openshift.io/cluster-api-machine-type: <node_role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_ID>-<node_role>-gpu-0
+    spec:
+      lifecycleHooks: {}
+      metadata: {}
+      providerSpec:
+        value:
+          apiVersion: openstackproviderconfig.openshift.io/v1alpha1
+          cloudName: openstack
+          cloudsSecret:
+            name: openstack-cloud-credentials
+            namespace: openshift-machine-api
+          flavor: <nova_gpu_flavor>
+          image: <glance_image_name_or_location>
+          kind: OpenstackProviderSpec
+          metadata:
+            creationTimestamp: null
+          networks:
+          - filter: {}
+            subnets:
+            - filter:
+                name: <infrastructure_ID>-nodes
+                tags: openshiftClusterID=<infrastructure_ID>
+          securityGroups:
+          - filter: {}
+            name: <infrastructure_ID>-<node_role>
+          serverGroupName: <infrastructure_ID>-<node_role>
+          serverMetadata:
+            Name: <infrastructure_ID>-<node_role>
+            openshiftClusterID: <infrastructure_ID>
+          tags:
+          - openshiftClusterID=<infrastructure_ID>
+          trunk: true
+          userDataSecret:
+            name: <node_role>-user-data
+```
+
+Save the file, and create machineset:
+
+```console
+$ oc create -f vgpu_machineset.yaml
+```
+
+And wait for new node to show up.
```

Comment:
Show up where? I guess in `oc get nodes`?

Reply:
yeah.